synapse/media/url_previewer.py

```diff
@@ -240,15 +240,14 @@ async def _do_preview(self, url: str, user: UserID, ts: int) -> bytes:
         cache_result = await self.store.get_url_cache(url, ts)
         if (
             cache_result
-            and cache_result["expires_ts"] > ts
-            and cache_result["response_code"] / 100 == 2
+            and cache_result.expires_ts > ts
+            and cache_result.response_code / 100 == 2
```

Comment:
This looks like it was meant to capture response codes in the range 200 to 299 inclusive. It would have done during Python 2; in Python 3 it only captures 200.

Reply:
Good-catch, care if I just drive by fix it?

Reply:
(Although: do any of the 2xx series responses other than 200 make sense here?)

Reply:
> This looks like it was meant to capture response codes in the range 200 to 299 inclusive. It would have done so during Python 2;

See https://github.com/matrix-org/synapse/pull/688, at which point it looks like we were definitely python 2.

Reply:
> (Although: do any of the 2xx series responses other than 200 make sense here?)

The non-cached path definitely treats 200-299 the same.